Pagination implemented in method that fetch the consumers of an eventType

ahmetcihank opened the pull request:
in terms of  #28 pagination is implemented and fixed tests accordingly  , pagination limit is considered for 10000 for now. I tested in my local environment and it is working.

aliok commented on backstage/plugins/knative-event-mesh-backend/src/providers/knativeEventMeshProcessor.ts:
```diff
@@ -14,6 +14,7 @@ import {TypeKnativeEvent} from "./types";
 export class KnativeEventMeshProcessor implements CatalogProcessor {
     private readonly catalogApi:CatalogClient;
     private readonly logger:Logger;
+    private readonly queryEntityPageLimit:number = 10000;
```
For making this testable, can you pass the `queryEntityPageLimit` in the constructor? It can be an optional argument, defaulting to 10000.

aliok commented on backstage/plugins/knative-event-mesh-backend/src/providers/knativeEventMeshProcessor.test.ts:
```diff
@@ -30,6 +30,8 @@ describe('KnativeEventMeshProcessor', () => {
                         'metadata.namespace':string,
                         'metadata.annotations.backstage.io/kubernetes-id':string,
                     },
+                    cursor: string,
+                    limit: number
```
Can you add a new test case?

The new test case should do these:
- `limit=5`
- Return 5 items in the mock API call, also return a cursor
- Test if the processor makes a 2nd call, with the cursor

aliok commented:
I actually noticed that there are some compilation errors.

They don't show up in GitHub actions. See https://github.com/knative-extensions/backstage-plugins/issues/45

Can you run `yarn tsc` locally and check out the compilation errors?

Other than that, I would actually change the `query` in the testcase definition to `queries` and define multiple mock calls.

Adding an inline comment now.

aliok commented on backstage/plugins/knative-event-mesh-backend/src/providers/knativeEventMeshProcessor.test.ts:
```diff
@@ -169,23 +182,108 @@ describe('KnativeEventMeshProcessor', () => {
+                        type: 'eventType',
+                    },
+                },
+                query: {
+                    queryEntitiesRequest: {
+                        filter: {
+                            kind: 'component',
+                            'metadata.namespace': 'default',
+                            'metadata.annotations.backstage.io/kubernetes-id': 'consumer-1',
+                        },
+                        cursor: undefined,
+                        limit: 5,
+                    },
+                    queryEntitiesResult:
+                        {
+                          items: [
+                                {
```
Maybe it makes more sense to be able to pass multiple queries that we want to mock. Something like:

```
    queries: [
       {
            queryEntitiesRequest: {
                filter: {...},
                cursor: undefined,
                limit: 5,
            },
            queryEntitiesResult: { 
                items: [...5 items],
                pageInfo: {
                    nextCursor: "AAAA",
                    prevCursor: "0"
               }
           }
        },
        {
            queryEntitiesRequest: {
                filter: {...},
                cursor: "AAAA",
                limit: 5,
            },
            queryEntitiesResult: { 
                items: [...2 items],
                pageInfo: {
                    nextCursor: undefined,
                    prevCursor: "AAAA"
               }
           }
        }
```

So, basically, in the suggestion above, I say that I want to mock 2 requests:
1. First receives 5 items and the cursor "AAAA"
2. Second one asks with the cursor "AAAA" and it receives 2 more items, with no next page cursor

What do you think?

Leo6Leo commented:
> Also, can you make the required things for the CLA?

Here is a short intro on what is CLA and how to do that, you may find this helpful! @ahmetcihank 
https://knative.dev/blog/articles/getting-started-blog-p2/#cla

aliok commented:
/lgtm
/approve

I know this took a bit long, thanks very much for your addressing comments patiently!

Can you fix the CLA issue? Then we can merge this one!

ahmetcihank commented:
> /lgtm /approve
> 
> I know this took a bit long, thanks very much for your addressing comments patiently!
> 
> Can you fix the CLA issue? Then we can merge this one!

CLA is fixed , It is my pleasure , to keep up with your comments.. Thanks a lot.

aliok reviewed (approved):
/lgtm
/approve

Thanks
